dist/datasource.js

```diff
@@ -204,7 +211,13 @@ function (angular, _, dateMath, moment) {
                 return timeshift(options.targets[0], options, targetsByRefId, datasourceSrv, metaTarget.outputMetricName)
             }
             else{
-                return ds.query(options)
+                var ds_res = ds.query(options);
+                if(ds_res.then){
+                    return ds_res;
+                }
+                else{
+                    return ds_res.toPromise();
+                }
```

Comment:
unrelated changes, there is another PR here https://github.com/GoshPosh/grafana-meta-queries/pull/132/files

Reply:
lets keep this PR specific to this change, will look into reviewing other one as well

Reply:
Thanks for taking a look. 

Quick question: Can you provide any kind of prediction as to when this change might become generally available? We have a broken 'datasource' currently and I really don't want to rollback by Grafana version. My users are also looking for alternatives to MetaQuery ...

Reply:
have asked the original author to simplify PR https://github.com/GoshPosh/grafana-meta-queries/pull/128, if I do not hear back in a day, will create a branch from author's commit and merge it

Reply:
@ryanpxyz https://github.com/GoshPosh/grafana-meta-queries/pull/128 should solve here right ?

Reply:
> lets keep this PR specific to this change, will look into reviewing other one as well

Oh I didn't see that. I will work on it in a few hours

Reply:
@Gauravshah For you information: We are currently running - 

Grafana v8.3.4
Grafana Operator v4.1.1

We get this error message from 'MetaQueries': "'Datasource [object Object] was not found'"

I am hoping that PR #141 fixes this.

Reply:
@Gauravshah changes done